google/cloud/aiplatform/metadata/resource.py

```diff
@@ -69,23 +77,32 @@ def __init__(
             project=project, location=location, credentials=credentials,
         )
 
-        # If we receive a full resource name, we extract the metadata_store_id and use that
-        if "/" in resource_name:
-            metadata_store_id = _Resource._extract_metadata_store_id(
-                resource_name, self._resource_noun
-            )
+        if resource:
+            self._gca_resource = resource
+            return
 
-        full_resource_name = utils.full_resource_name(
-            resource_name=resource_name,
-            resource_noun=f"metadataStores/{metadata_store_id}/{self._resource_noun}",
-            project=self.project,
-            location=self.location,
-        )
+        full_resource_name = resource_name
+        # Construct the full_resource_name if input resource_name is the resource_id
+        if "/" not in resource_name:
+            full_resource_name = utils.full_resource_name(
+                resource_name=resource_name,
+                resource_noun=f"metadataStores/{metadata_store_id}/{self._resource_noun}",
+                project=self.project,
+                location=self.location,
+            )
 
         self._gca_resource = getattr(self.api_client, self._getter_method)(
             name=full_resource_name
         )
 
+    @property
+    def metadata(self) -> Dict:
+        return dict(self._gca_resource.metadata.items())
```

Comment:
Please add TODO to update with json_format please.

Reply:
done.